kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/PodOperationContext.java

```diff
@@ -164,10 +164,11 @@ public void addQueryParameters(URLBuilder httpUrlBuilder) {
     if (in != null || redirectingIn) {
       httpUrlBuilder.addQueryParameter("stdin", "true");
     }
-    if (output != null) {
+    boolean debug = ExecWebSocketListener.LOGGER.isDebugEnabled();
```

Comment:
This reveals an odd coupling between ExecWebSocketListener and this class.

The name of the method suggests a wider usage of these parameters, however, they are only ultimately consumed at the `public ExecWatch exec(String... command)` method. Maybe we should rename the addQueryParameters method.

Reply:
> This reveals an odd coupling between ExecWebSocketListener and this class.

It's partly an issue with the existing code structure.  The entry class is PodOperationsImpl, but it's in a different package from the rest of the logic - presumably because there could eventually be a v2.  That's how most of dsl.internal is used.

> Maybe we should rename the addQueryParameters method.

It's usage should be clear from the context :) You could also make the case that the classes in dsl.internal should move into core.v1 and have methods be made package private.